File: DirectoryListing/app/src/main/java/com/example/directorylisting/helpers/FileHelper.java
/*
 * Copyright © 2017 Michael Steele.
 * All rights reserved.
 */

package com.example.directorylisting.helpers;

import android.content.Context;
import android.util.Log;

import com.example.directorylisting.api.WebService;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.InputStream;
import java.io.OutputStream;
import java.io.OutputStreamWriter;
import java.io.Writer;

public class FileHelper {

    public static File cachedFile(Context context, String fileName) {
        return new File(context.getCacheDir(), fileName);
    }

    public static void readFile(File file, WebService.BasicSuccessObjectFailureInterface basicSuccessObjectFailureInterface) {
        InputStream inputStream = null;
        try {
            inputStream = new FileInputStream(file);

            basicSuccessObjectFailureInterface.onResponse(inputStream);
        } catch (Exception e) {
            Log.e(WebService.class.toString(), "Failed to read cert.");
            e.printStackTrace();
        }
    }

    public static void writeFile(Context context, File file, String text, WebService.BasicSuccessFailureInterface basicSuccessFailureInterface) {
        try {
            OutputStream outputStream = new FileOutputStream(file);
            Writer writer = new OutputStreamWriter(outputStream);

            writer.write(text);

            writer.close();

            basicSuccessFailureInterface.onSuccess();
        } catch (Exception e) {
            Log.e(WebService.class.toString(), "Failed to write downloaded file.");

            e.printStackTrace();

            basicSuccessFailureInterface.onFailure();
        }
    }

}

File: DirectoryListing/app/src/main/java/com/example/directorylisting/helpers/FileHelper.kt
/*
 * Copyright © 2017 Michael Steele.
 * All rights reserved.
 */

package com.example.directorylisting.helpers

import android.content.Context
import android.util.Log
import com.example.directorylisting.api.WebService
import java.io.*

object FileHelper {

    fun cachedFile(context: Context, fileName: String): File {
        return File(context.cacheDir, fileName)
    }

    fun readFile(file: File, basicSuccessObjectFailureInterface: WebService.BasicSuccessObjectFailureInterface) {
        var inputStream: InputStream? = null
        try {
            inputStream = FileInputStream(file)

            basicSuccessObjectFailureInterface.onResponse(inputStream)
        } catch (e: Exception) {
            Log.e(WebService::class.java.toString(), "Failed to read cert.")
            e.printStackTrace()
        }

    }

    fun writeFile(context: Context, file: File, text: String, basicSuccessFailureInterface: WebService.BasicSuccessFailureInterface) {
        try {
            val outputStream = FileOutputStream(file)
            val writer = OutputStreamWriter(outputStream)

            writer.write(text)

            writer.close()

            basicSuccessFailureInterface.onSuccess()
        } catch (e: Exception) {
            Log.e(WebService::class.java.toString(), "Failed to write downloaded file.")

            e.printStackTrace()

            basicSuccessFailureInterface.onFailure()
        }

    }

}
